pyomo/core/base/set.py

```diff
@@ -205,20 +307,44 @@ def __call__(self, parent, idx):
     def constant(self):
         return self._A.constant() and self._B.constant()
 
-class RangeSetInitializer(InitializerBase):
+    def contains_indices(self):
+        return self._A.contains_indices() or self._B.contains_indices()
+
+    def indices(self):
+        if self._A.contains_indices():
+            if self._B.contains_indices():
+                if set(self._A.indices()) != set (self._B.indices()):
+                    raise ValueError(
+                        "SetIntersectInitializer contains two "
+                        "sub-initializers with inconsistent external indices")
+            return self._A.indices()
+        else:
+            # It is OK (and desirable) for this to raise the exception
+            # if B does not contain external indices
+            return self._B.indices()
+
+class BoundsInitializer(InitializerBase):
```

Comment:
Can you add some very brief class documentation to describe what these initializer classes are used for? I think it will be useful for developers in the future.